Objects/typeobject.c

```diff
@@ -3152,6 +3152,18 @@ PyType_FromSpec(PyType_Spec *spec)
     return PyType_FromSpecWithBases(spec, NULL);
 }
 
+PyObject *
+PyType_GetName(PyTypeObject *type)
+{
+    assert(PyType_Check(type));
+    if (_PyType_HasFeature(type, Py_TPFLAGS_HEAPTYPE)) {
+        PyHeapTypeObject *ht_type = (PyHeapTypeObject *)type;
+        Py_INCREF(ht_type->ht_name);
+        return ht_type->ht_name;
+    }
+    return PyUnicode_FromString(_PyType_Name(type));
+}
+
```

Comment:
Would it work to call the existing function `type_name` instead?

Reply:
Of course. Updated :).